### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by opengs** [TooTallNate/node-socks-proxy-agent#93]
Socks-proxy-agent leaks memory on every creation.

Here is example code that leaks memory very fast
```
const SocksProxyAgent = require('socks-proxy-agent')

function makeAgent() {
  setTimeout(makeAgent, 5)
  return new SocksProxyAgent.SocksProxyAgent({
    hostname: 'google.com'
  })
}

makeAgent()
```

**Comment by opengs** [comment]
No memory leak with custom dnsCache

```
const SocksProxyAgent = require('socks-proxy-agent')

const CacheableLookup = require('cacheable-lookup');

const cacheable = new CacheableLookup();

function makeAgent() {
  setTimeout(makeAgent, 5)
  return new SocksProxyAgent.SocksProxyAgent({
    hostname: 'google.com'
  }, {dnsCache: cacheable})
}

makeAgent()
```

**Comment by Kikobeats** [comment]
which version do you test?

**Comment by opengs** [comment]
@Kikobeats in my package json I have ```"socks-proxy-agent": "^6.2.0-beta.0"```

after i type ```yarn list --pattern socks-proxy-agent```
and it says ```socks-proxy-agent@6.2.0-beta.0```

**Comment by Kikobeats** [comment]
can you test using `v6.2.0-beta.1`? I think it resolves the thing.

**Comment by opengs** [comment]
@Kikobeats there is no problem in `v6.2.0-beta.1` . May this issue be opened until v6.2.1 (or next stable) release, so it can be automatically installed when i type `yarn add socks-proxy-agent` :) ?

**Comment by Kikobeats** [comment]
great; published `socks-proxy-agent@6.2.0` 🙂
